backend/routes/quests.go

```diff
@@ -211,6 +179,68 @@ func GetMainUserQuests(w http.ResponseWriter, r *http.Request) {
 	routeutils.WriteDataJson(w, string(quests))
 }
 
+func GetMainUserQuestsProgress(w http.ResponseWriter, r *http.Request) {
+	userAddress := r.URL.Query().Get("address")
+	if userAddress == "" {
+		routeutils.WriteErrorJson(w, http.StatusBadRequest, "Missing address parameter")
+		return
+	}
+
+	_, err := core.PostgresQueryJson[UserMainQuest]("SELECT questId FROM UserMainQuests where user_address = $1", userAddress)
```

Comment:
You'll want to query from `MainQuests` here to get the id and quest type.

Reply:
why do we need to do this,  MainQuests has no relation to user address that we can query base on WHERE clause

even if we query MainQuests, we only need questId which we are getting from UserMainQuests, and type is of no use,

or you plan to add the type inside the retuern data @b-j-roberts

Reply:
The UserMainQuests only stores data marking that a user completed a quest. It doesn't store any quest information if a user hasn't completed a quest.

This route is intended to get the users progress on all quests, not completed ones. So querying `MainQuests` and `DailyQuests` will list all quest ids and types.
The type will be used to help determine how to compute the progress. Which will be done in the future. For now the hardcoded progress values are fine

Reply:
@b-j-roberts can you check above

Reply:
Sorry, what do you mean?

Reply:
questId on MainQuests is key right ? and type is the quest_type